show/plugins/barefoot.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python
+
+import click
+import json
+import subprocess
+from sonic_py_common import device_info
+
+@click.group()
+def barefoot():
+    pass
+
+@barefoot.command()
+def profile():
+    # Check if profile can be changed
+    completed_process = subprocess.run(['docker', 'exec', '-it', 'syncd',
+        'test', '-h', '/opt/bfn/install'])
+    if completed_process.returncode != 0:
+        click.echo('Current profile: default')
+        return
+    
+    # Get chip family
+    hwsku_dir = device_info.get_path_to_hwsku_dir()
+    with open(hwsku_dir + '/switch-tna-sai.conf') as file:
+        chip_family = json.load(file)['chip_list'][0]['chip_family']
+    
+    # Print current profile
+    click.echo('Current profile: ', nl=False)
+    subprocess.run('docker exec -it syncd readlink /opt/bfn/install | sed '
+        's/install_\\\\\\(.\\*\\\\\\)_profile/\\\\1/', check=True, shell=True)
+    
+    # Exclude current and unsupported profiles
+    opts = ''
+    if chip_family == 'Tofino':
+        opts = '\\! -name install_y\\*_profile '
+    elif chip_family == 'Tofino2':
+        opts = '\\! -name install_x\\*_profile '
+    
+    # Print profile list
+    click.echo('Available profile(s):')
+    subprocess.run('docker exec -it syncd find /opt/bfn -mindepth 1 '
+        '-maxdepth 1 -type d -name install_\\*_profile ' + opts + '| sed '
+        's%/opt/bfn/install_\\\\\\(.\\*\\\\\\)_profile%\\\\1%', shell=True)
```

Comment:
just FYI, raw string will reduce escape sequence length. Example:
```
r'\\hello\\'  instead of '\\\\hello\\\\'
```
https://www.journaldev.com/23598/python-raw-string

Reply:
Thanks for your review.
Raw strings are now used where appropriate.